### ISSUE TITLE
Issue with adding middleware to specific route

### ISSUE THREAD

**Issue by david123-mtl** [VilledeMontreal/express-idempotency-mongo-adapter#7]
**Issue** : 
When I use it directly with `app.use`, the code works fine.
```
if (app) {
        let adapter;
        if (!configs.testingMode && !configs.idempotencyMiddleware.adapter.useMemoryAdapter) {
            adapter = newAdapter({
                config: {
                    uri: configs.mongo.connectionString
                },
                collectionPrefix: configs.idempotencyMiddleware.adapter.collectionPrefix,
                ttl: configs.idempotencyMiddleware.adapter.ttl
            });
        }
        app.use(
            idempotency({
                dataAdapter: adapter,
                idempotencyKeyHeader: configs.idempotencyMiddleware.keyHeader
            })
        );
    }
```
but when I used it to initialize the middleware directly with the route, 
```
import { idempotency } from "express-idempotency";
import { newAdapter } from "express-idempotency-mongo-adapter";
// ...
{
    method: HttpMethods.GET,
    path: 'my-route',
    handler: rdeSpdvController.getProductClasses,
    middlewares:  [idempotency({
        dataAdapter: adapter,
        idempotencyKeyHeader: configs.idempotencyMiddleware.keyHeader
    })]
},
```
I kept getting this error : 
```
/home/vm/projects/pet-licensing-api/node_modules/reflect-metadata/Reflect.ts:1090
            const decorated = decorator(target);
                              ^
TypeError: decorator is not a function
    at DecorateConstructor (/home/vm/projects/pet-licensing-api/node_modules/reflect-metadata/Reflect.ts:1090:31)
    at Object.decorate (/home/vm/projects/pet-licensing-api/node_modules/reflect-metadata/Reflect.ts:284:20)
    at __decorate (/home/vm/projects/pet-licensing-api/node_modules/express-idempotency-mongo-adapter/dist/adapter.js:5:92)
    at Object.<anonymous> (/home/vm/projects/pet-licensing-api/node_modules/express-idempotency-mongo-adapter/dist/adapter.js:217:16)
    at Module._compile (internal/modules/cjs/loader.js:778:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:789:10)
    at Module.load (internal/modules/cjs/loader.js:653:32)
    at tryModuleLoad (internal/modules/cjs/loader.js:593:12)
    at Function.Module._load (internal/modules/cjs/loader.js:585:3)
    at Module.require (internal/modules/cjs/loader.js:692:17)
    at require (internal/modules/cjs/helpers.js:25:18)
    at Object.<anonymous> (/home/vm/projects/pet-licensing-api/node_modules/express-idempotency-mongo-adapter/dist/index.js:6:10)
    at Module._compile (internal/modules/cjs/loader.js:778:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:789:10)
    at Module.load (internal/modules/cjs/loader.js:653:32)
    at tryModuleLoad (internal/modules/cjs/loader.js:593:12)
    at Function.Module._load (internal/modules/cjs/loader.js:585:3)
    at Module.require (internal/modules/cjs/loader.js:692:17)
    at require (internal/modules/cjs/helpers.js:25:18)
    at Object.<anonymous> (/home/vm/projects/pet-licensing-api/src/routes/rdeSpdv.ts:3:1)
    at Module._compile (internal/modules/cjs/loader.js:778:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:789:10)
    at Module.load (internal/modules/cjs/loader.js:653:32)
    at tryModuleLoad (internal/modules/cjs/loader.js:593:12)
    at Function.Module._load (internal/modules/cjs/loader.js:585:3)
    at Module.require (internal/modules/cjs/loader.js:692:17)
    at require (internal/modules/cjs/helpers.js:25:18)
    at Object.<anonymous> (/home/vm/projects/pet-licensing-api/src/routes.ts:16:1)
    at Module._compile (internal/modules/cjs/loader.js:778:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:789:10)
    at Module.load (internal/modules/cjs/loader.js:653:32)
    at tryModuleLoad (internal/modules/cjs/loader.js:593:12)
    at Function.Module._load (internal/modules/cjs/loader.js:585:3)
    at Module.require (internal/modules/cjs/loader.js:692:17)
```
**Proposed solution** : 
Move `autobind-decorator` to `depedencies` seems to solve the issue

**Comment by joscelynjean** [comment]
Hi, thank you for your comment.

It seems that you have exactly the same error from this issue in the idempotency express middleware.
https://github.com/VilledeMontreal/express-idempotency/issues/10

I will fix it.

Thank you for your feedback.

**Comment by joscelynjean** [comment]
I modified the TTL example to show how to use it for a specific route instead on all route (app.use).

It will be included in the release 1.0.2.
